core/src/main/java/org/apache/calcite/rel/rules/AggregateReduceFunctionsRule.java

```diff
@@ -608,8 +609,31 @@ private static RexNode reduceStddev(
             aggCallMapping,
             oldAggRel.getInput()::fieldIsNullable);
 
+    final RexNode avgSumSquaredArg =
+        rexBuilder.makeCall(pos, SqlStdOperatorTable.DIVIDE, sumSquaredArg, countArg);
+    final RexNode diff =
+        rexBuilder.makeCall(pos, SqlStdOperatorTable.MINUS, sumArgSquared, avgSumSquaredArg);
+
+    final RelDataType oldArgType =
+        SqlTypeUtil.projectTypes(oldAggRel.getInput().getRowType(), oldCall.getArgList())
+            .get(0);
+    final RexNode correctedDiff;
+
+    switch (oldArgType.getSqlTypeName()) {
+    case DOUBLE: case DECIMAL:
+      final RexNode zeroLiteral =
+          rexBuilder.makeExactLiteral(BigDecimal.ZERO);
+      correctedDiff =
+          rexBuilder.makeCall(SqlStdOperatorTable.CASE,
+              rexBuilder.makeCall(SqlStdOperatorTable.LESS_THAN, diff, zeroLiteral),
```

Comment:
I wonder whether using less than or equal would make sense
doubles have this concept of a negative zero, which may cause problems
I am not sure whether it can surface here, but there should be no harm